src/plugins/usage_collection/public/plugin.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Reporter, METRIC_TYPE } from '@kbn/analytics';
+// import { Storage } from 'ui/storage';
+import { Storage } from '../../kibana_utils/public';
+import { createReporter } from './services';
+import {
+  PluginInitializerContext,
+  Plugin,
+  CoreSetup,
+  CoreStart,
+  HttpServiceBase,
+} from '../../../core/public';
+
+interface PublicConfigType {
+  uiMetric: {
+    enabled: boolean;
+    debug: boolean;
+  };
+}
+
+export interface UsageCollectionSetup {
+  allowTrackUserAgent: (allow: boolean) => void;
+  reportUiStats: Reporter['reportUiStats'];
+  METRIC_TYPE: typeof METRIC_TYPE;
+}
+
+export function isUnauthenticated(http: HttpServiceBase) {
+  const { basePath } = http;
+  const path = basePath.remove(window.location.pathname);
+  return path === '/login' || path === '/logout' || path === '/logged_out' || path === '/status';
+}
```

Comment:
use `http.anonymousPaths.isAnonymous(window.location.pathname)` instead

Reply:
does this `http.anonymousPaths.isAnonymous` return `true` if the paths are `/login`, `/logout`, `/logged_out`, or `/status` ?

Reply:
It returns true **only** if the route is anonymous.

Plug-ins register their anonymous routes. But yes, atm the routes you listed are exactly what is registered as anon.

(except for the /status that can be anonymous or authenticated depending on settings, which is one of the reason to use the core utility instead of re-implementing the method)

https://github.com/elastic/kibana/blob/38c17d6c7d03c2697f92f95acceb828fd513bb0c/src/plugins/status_page/public/plugin.ts#L24-L31